packages/@lwc/ssr-compiler/src/compile-template/index.ts

```diff
@@ -65,18 +70,65 @@ const bExportTemplate = esTemplate<
     }
 `;
 
-export default function compileTemplate(src: string, filename: string, options: TransformOptions) {
-    const { root, warnings } = parse(src);
+function templateUsesDirective(root: Root, target: ElementDirective['name']): boolean {
```

Comment:
This traverses the full AST; can we move this check to while we're doing the rest of the parsing?